File: Pub-Sub-Pulsar/vendas/src/main/java/com/example/pulsar/vendas/controller/SaleController.java
package com.example.pulsar.vendas.controller;

import com.example.pulsar.vendas.model.Saler;
import com.example.pulsar.vendas.service.PubService;
import org.apache.pulsar.client.api.PulsarClientException;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/sale")
@SuppressWarnings("unchecked")
public class SaleController {

    @Autowired
    private PubService pubService;

    @PostMapping
    public ResponseEntity<String> newSale (@RequestBody Saler saler) throws PulsarClientException {
        pubService.publishSalerEvent(saler);
        return new ResponseEntity("Sale made successfully!", HttpStatus.OK);
    }
}

File: SQS-SNS/vendas/src/main/kotlin/com/sns/service/vendas/controller/SaleController.kt
package com.sns.service.vendas.controller

import com.sns.service.vendas.model.Saler
import com.sns.service.vendas.service.SnsService
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController


@RestController
@RequestMapping("/sale")
class SaleController(
    val snsService: SnsService
) {

    @PostMapping
    fun newSale(@RequestBody saler: Saler): ResponseEntity<String> {
        snsService.publishSalerEvent(saler)
        return ResponseEntity("Sale made successfully!", HttpStatus.OK)
    }

}